src/MetricsProvider.ts

```diff
@@ -0,0 +1,132 @@
+import { COUNTLY_API_URL } from './config'
+import type { consentTypes, eventTypes } from 'countly-sdk-web'
+import Countly from 'countly-sdk-web'
+
+interface MetricsProviderConstructorOptions {
+  appKey: string
+  url?: string
+  autoTrack?: boolean
+}
+
+export default class MetricsProvider {
+  private readonly minimalEvents: eventTypes[] = ['sessions', 'views']
+  private readonly marketingEvents: eventTypes[] = ['attribution', 'users', 'location']
+  private readonly trackingEvents: eventTypes[] = ['events', 'crashes', 'apm']
+  private readonly performanceEvents: eventTypes[] = ['scrolls', 'clicks', 'forms', 'star-rating', 'feedback']
+  private readonly groupedFeatures: Record<consentTypes, eventTypes[]> = {
+    all: [
+      ...this.marketingEvents,
+      ...this.minimalEvents,
+      ...this.performanceEvents,
+      ...this.trackingEvents
+    ],
+    marketing: this.marketingEvents,
+    minimal: this.minimalEvents,
+    performance: this.performanceEvents,
+    tracking: this.trackingEvents
+  }
+
+  private sessionStarted: boolean = false
+  private readonly _consentGranted: Set<consentTypes> = new Set()
+
+  constructor ({ autoTrack = true, url = COUNTLY_API_URL, appKey }: MetricsProviderConstructorOptions) {
+    this.metricsService.init({
+      app_key: appKey,
+      url: url,
+      require_consent: true
+    })
+
+    this.metricsService.group_features(this.groupedFeatures)
+
+    if (autoTrack) {
+      this.setupAutoTrack()
+    }
+  }
+
+  get metricsService (): typeof Countly {
+    return Countly
+  }
+
+  get consentGranted (): consentTypes[] {
+    return [...this._consentGranted]
+  }
+
+  setupAutoTrack () {
+    this.metricsService.track_clicks()
+    this.metricsService.track_errors()
+    this.metricsService.track_forms()
+    this.metricsService.track_links()
+    this.metricsService.track_pageview()
+    this.metricsService.track_scrolls()
+    this.metricsService.track_sessions()
+    this.metricsService.track_view()
+  }
+
+  addConsent (consent: consentTypes | consentTypes[]) {
+    if (!Array.isArray(consent)) {
+      consent = [consent]
+    }
+    consent.forEach(c => this._consentGranted.add(c))
+    this.metricsService.add_consent(consent)
+  }
+
+  removeConsent (consent: consentTypes | consentTypes[]) {
+    if (!Array.isArray(consent)) {
+      consent = [consent]
+    }
+    consent.forEach(c => this._consentGranted.delete(c))
+    this.metricsService.remove_consent(consent, true)
+  }
+
+  checkConsent (consent: consentTypes) {
+    if (!(consent in this.groupedFeatures)) {
+      throw new Error('Unknown consent type')
+    }
```

Comment:
this is not true. see https://github.com/Countly/countly-sdk-web/issues/337. we should allow check_consent for our groups, and countly specific features.

Reply:
```suggestion
    const featuresArray = this.groupedFeatures[consent]
    if (featuresArray == null) {
      return this.metricsService.check_consent(consent)	
    }
```